components/org.wso2.carbon.identity.scim.common/src/main/java/org/wso2/carbon/identity/scim/common/utils/SCIMCommonUtils.java

```diff
@@ -161,4 +179,48 @@ public static String getPrimaryFreeGroupName(String groupName) {
         return groupName;
     }
 
+    /**
+     * Set SCIM attributes for super tenant admin users.
+     */
+    public static void setAdminSCIMAttributes() {
+
+        try {
+            int superTenantId = IdentityTenantUtil.getTenantId(SUPER_TENANT_DOMAIN);
+            UserStoreManager userStoreManager =
+                    (UserStoreManager) SCIMCommonComponentHolder.getRealmService().
+                            getTenantUserRealm(superTenantId).getUserStoreManager();
+
+            if (userStoreManager.isSCIMEnabled()) {
+                //get admin user name from claim utils
+                String adminUsername = ClaimsMgtUtil.getAdminUserNameFromTenantId(IdentityTenantUtil.getRealmService(),
+                        superTenantId);
+                Map<String, String> claimsList = new HashMap<>();
+
+                //get scim id attribute. generate new metadata if null
+                String scim_id = userStoreManager.getUserClaimValue(adminUsername, SCIMConstants.ID_URI,
+                        UserCoreConstants.DEFAULT_PROFILE);
+                if (StringUtils.isEmpty(scim_id)) {
+                    String id = UUID.randomUUID().toString();
+                    claimsList.put(SCIMConstants.ID_URI, id);
+                    claimsList.put(SCIMConstants.USER_NAME_URI, adminUsername);
+
+                    Date date = new Date();
+                    String createdDate = formatDateTime(date);
+                    claimsList.put(SCIMConstants.META_CREATED_URI, createdDate);
+                    claimsList.put(SCIMConstants.META_LAST_MODIFIED_URI, createdDate);
+                    userStoreManager.setUserClaimValues(adminUsername, claimsList, UserCoreConstants.DEFAULT_PROFILE);
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Its better to catch the particular exception rather than catching the generic exception. Or is it thrown by some downstream methods.?, if that is the case then its fine.

Furthermore, AFAIU this is not the top level of the code, so better to throw the exception here rather than only loging it. It should be catched at the top level of the code and handled there.

Reply:
ClaimsMgtUtil.getAdminUserNameFromTenantId throws a generic Exception hence we need to catch the generic exception here. I have modified the exception to be handled at top level of the code in #76